flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/join/SortMergeJoinOperator.java

```diff
@@ -158,16 +140,26 @@ public void open() throws Exception {
 		this.memManager = this.getContainingTask().getEnvironment().getMemoryManager();
 		this.ioManager = this.getContainingTask().getEnvironment().getIOManager();
 
+		long totalMemory = computeMemorySize();
+		long totalSortMem = totalMemory - externalBufferMemory;
+		if (totalSortMem < 0) {
+			throw new TableException("Memory size is too small: " +
+					totalMemory + ", please increase manage memory of task manager.");
+		}
+
+		long sortMemory1 = (long) (totalSortMem * leftMemRatio);
```

Comment:
Should we also check the minimum memory for sorter?

Reply:
In `BinaryExternalSorter`, will do this check.
Actually here, we don't know the minimum memory of algorithm.